### PR TITLE
Coverage and Quality Report: Add reporting on use of forced-colors media query and currentcolor value

### PR DESCRIPTION
**Pull request by jongund**
The pull request updates the coverage script to:
1. Remove information about the use of `forced-color-adjust` from the report, since I thin we should not even be using it anymore in examples.
2. Adds information on examples using `forced-colors` media query.
3. Adds information on examples using `currentcolor` value for CSS properties that support color values.

[Preview of Updated Coverage Report](https://deploy-preview-324--aria-practices.netlify.app/aria/apg/about/coverage-and-quality/)

___
[WAI Preview Link](https://deploy-preview-324--aria-practices.netlify.app/ARIA/apg) _(Last built on Tue, 23 Jul 2024 06:01:44 GMT)._

### CODE REVIEW

**Comment by mcking65** [comment]
@jongund is this a replacement for #2902

**Comment by jongund** [comment]
@mcking65 
Yes this would be the report we would want now.
Writing the high contrast support practices clarified for me the relevant information we needed in the report.

File: scripts/coverage-report.js [hunk 1]
```diff
@@ -521,7 +521,9 @@ glob
       contentCSS: getNumberOfReferences(dataCSS, 'content'),
       beforeCSS: getNumberOfReferences(dataCSS, '::before'),
       afterCSS: getNumberOfReferences(dataCSS, '::after'),
-      forcedColorAdjust: getNumberOfReferences(dataCSS, 'forced-color-adjust'),
+      //      forcedColorAdjust: getNumberOfReferences(dataCSS, 'forced-color-adjust'),
```
Comment (howard-e) [inline scripts/coverage-report.js:524]: Could this be removed instead of commenting out?

File: scripts/coverage-report.js [hunk 2]
```diff
@@ -1008,7 +1010,9 @@ let IndexOfExampleGraphics = indexOfExamples.reduce(function (set, example) {
   let count = example.svgHTML;
   count += example.svgCSS;
   count += example.svgJS;
-  count += example.forcedColorAdjust;
+  //  count += example.forcedColorAdjust;
```
Comment (howard-e) [inline scripts/coverage-report.js:1013]: Same question as above since it should no longer be used.

File: scripts/coverage-report.js [hunk 3]
```diff
@@ -1125,7 +1140,9 @@ $('#example_mouse_pointer_tbody').html(IndexOfExampleMousePointer);
 $('#example_summary_total').html(indexOfExamples.length);
 $('#example_summary_hc').html(countHighContrast);
 $('#example_summary_svg').html(countSVG);
-$('#example_summary_force_color').html(countForcedColorAdjust);
+$('#example_summary_force_colors').html(countForcedColors);
+$('#example_summary_current_color').html(countCurrentColor);
+// $('#example_summary_force_color_adjust').html(countForcedColorAdjust);
```
Comment (howard-e) [inline scripts/coverage-report.js:1145]: Same question as above

File: scripts/coverage-report.template [hunk 4]
```diff
@@ -168,9 +168,19 @@
...
+          <tr>
+            <th scope="row">Uses <code>currentColor</code> value</th>
+            <td id="example_summary_current_color"></td>
+          </tr>
+<!--
+          <tr>
+            <th scope="row">Uses <code>forced-color-adjust</code> in CSS/SVG</th>
+            <td id="example_summary_force_color_adjust"></td>
+          </tr>
+-->
```
Comment (howard-e) [inline scripts/coverage-report.template:183]: Same question as above

**Review by howard-e — COMMENTED** [review]
@jongund other than the questions in line, could you also merge this with `main` so https://github.com/w3c/aria-practices/pull/3026 changes could be included?

`npm run coverage-report` can be ran again afterwards to make sure things are in sync.

File: scripts/coverage-report.js (hunk 2 above)
Comment (jongund) [reply scripts/coverage-report.js:1013]: Yes, I can take it out

File: scripts/coverage-report.template (hunk 4 above)
Comment (jongund) [reply scripts/coverage-report.template:183]: I will take it out

**Comment by jongund** [comment]
@mcking65 
I also removed the CSS :before, :after and content columns in the table.  
Changed the name of the table to "SVG and High Contrast Techniques".  
The use of CSS :before and :after are not part of the coding practices.

**Comment by mcking65** [comment]
@howard-e 
Why is Check examples/index.html / coverage failing?

I assume that is why the preview still shows the CSS columns Jon removed from the graphics techniques table; the generation script didn't run?

I requested re-review now that Jon has made requested updates.

**Comment by howard-e** [comment]
> @howard-e Why is Check examples/index.html / coverage failing?

This PR needs to manually re-run `npm run coverage-report` because the lint-staged step to automatically do it doesn't trigger when the generating script alone is updated in a PR. Instead, it was only being triggered when there are changes to `content/patterns/**/examples/*.html`.

But that is odd that the generating scripts weren’t being captured so I submitted #3048 to address it.

> I assume that is why the preview still shows the CSS columns Jon removed from the graphics techniques table; the generation script didn't run?

Not exactly, this was because there was an invalid file read happening with the new experimental content support on the build repo because of an uncaught assumption that the newly created files in #2977 were already present when the build repo references w3c/aria-practices.

Already submitted https://github.com/w3c/wai-aria-practices/pull/330 to fix and has been merged and and Jon’s PR preview has been regenerated

> I requested re-review now that Jon has made requested updates.

Will check it!

**Review by howard-e — COMMENTED** [review]
@jongund thanks for addressing the feedback and the additional changes look good to me.

This still just needs a merge with `main` to include #3026, running `npm run coverage-report` and a commit and push to be ready.

**Pull request merged**